src/index/FTSAlgorithms.h

```diff
@@ -35,6 +35,10 @@ class FTSAlgorithms {
       const Index::WordEntityPostings& matchingContextsWep,
       const Index::WordEntityPostings& eBlockWep);
 
+  static Index::WordEntityPostings crossIntersect(
+      const Index::WordEntityPostings& matchingContextsWep,
+      const Index::WordEntityPostings& eBlockWep);
+
```

Comment:
Please add a comment what this does, and how it differs from `intersect`.
Is the point here, that the input to `crossIntersect` can have the same text multiple times because of different matching words, and if yes, what is the semantics?